### PR TITLE
[AC-9444] Add InnovationStage model

### PR DESCRIPTION
**Pull request by iankigen**
## [AC-9444](https://masschallenge.atlassian.net/browse/AC-9444)

**Changes introduced**
- add InnovationStage model

**Testing**
- On this branch, `cd path/to/accelerate && make migrate`
- As an admin user, visit http://localhost:8181/admin/
- Confirm that the def of done according to [AC-9444](https://masschallenge.atlassian.net/browse/AC-9444) is satisfied

[Sibling PR](https://github.com/masschallenge/accelerate/pull/3250)

### CODE REVIEW

File: accelerator/models/innovation_stage.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+from accelerator_abstract.models.accelerator_model import AcceleratorModel
+
+
+class InnovationStage(AcceleratorModel):
+    name = models.CharField(max_length=255)
+    description = models.TextField()
+    image = ImageField(
+        upload_to='innovation_stage_images', blank=True, null=True)
+    display_priority = models.PositiveIntegerField()
```
Comment (AmosWels) [inline accelerator/models/innovation_stage.py:12]: could pass `null=False`, or provide a default

Comment (iankigen) [reply accelerator/models/innovation_stage.py:12]: by default; blank=False & null=False

**Review by AmosWels — APPROVED** [review]
changes look good

**Review by shakirandagire — APPROVED** [review]
QA checks out as well.
✅  Model is created as per the definition of done and its migration exists
✅ Innovation stage model is added to the admin page
Screenshot 2022-04-21 at 13 05 14
✅ Add InnovationStageFactory with sensible default values
✅ add M2M relationship to:CoreProfile - innovation_stage_interest and Program - supported_innovation_stages
✅ supported_innovation_stages option is added to the [Program administration site](http://staging.masschallenge.org/admin/mc/program/)
Screenshot 2022-04-21 at 13 09 41

**Pull request merged**